src/utilities/utilities.jl

```diff
@@ -330,7 +330,7 @@ Allows to supply `f`, which gets applied to every point.
 """
 function matrix_grid(f, x::AbstractArray, y::AbstractArray, z::AbstractMatrix)
     g = map(CartesianIndices(z)) do i
-        return f(Point3f(get_dim(x, i, 1, size(z)), get_dim(y, i, 2, size(z)), z[i]))
+        return f(Point3(get_dim(x, i, 1, size(z)), get_dim(y, i, 2, size(z)), z[i]))
```

Comment:
This might return a `Point3{Int}` or similar in the right circumstance; do we want to allow that?

Reply:
I think more generally the question of "What do we want to convert to" is still open. For `plot.converted` we have 3 options:
1. Always convert to Float64 derived types
2. Keep whatever type comes from the first conversion 
3. Use generic Any types

1 is easy and what I set up for now. For GLMakie and WGLMakie it leads to extra conversions later down the line though, which I don't really like.

2 will require all the possible `plot.converted` types to be able to convert automatically. Currently at least `Rect` does not do that. It might also be bad for setting up empty plots, though I guess it's not that much of a problem.

3 would probably suck for performance?

Reply:
I've been pushing for lazy converts in my plot object refsctors (2)